add options parameter to add redoc options and theming capacity.

youenchene opened the pull request:
Small contributions to add a options parameter to handle redoc configuration : https://github.com/Redocly/redoc/blob/main/docs/config.md

Also added documentation for understanding quicker the different parameters of go-redoc.

wowissu committed:
feat: allow to read sub spec by SpecDir

wowissu committed:
feat: only allow .yaml and .json

wowissu committed:
feat: use cmd not local file

wowissu committed:
fix: misplaced go:embed directive

hohmannr committed:
feat: makes Makefile redocly update .PHONY, updates the redocly latest bundle url and executes an update

youenchene committed:
add options parameter to add redoc options and theming capacity.

mvrilo commented on redoc.go:
```diff
@@ -21,6 +23,7 @@ type Redoc struct {
 	SpecFS      *embed.FS
 	Title       string
 	Description string
+	Options     string
```
Not a big fan of expecting a json string here.
It also seems that `json.Valid` doesn't ensure the string is an actual object.
https://go.dev/play/p/gmBO0XamJob

I see the redoc's Config type is quite [extensive](https://github.com/Redocly/redoc/blob/124309592670979d1a647718e5ad94b9b9e56e48/src/services/RedocNormalizedOptions.ts#L14) - so supporting all of these in a struct type might be overkill.

So why not use a `map[string]any` as type then later `json.Marshal` it?
That would ensure we're always receiving an object.

Your example would look like this:

```go
Options: map[string]any{
  "disableSearch": true,
  "theme": map[string]any{
    "colors": `{"primary":{"main":"#297b21"}}`,
  },
}
```

mvrilo replied:
Also instead of naming it `Options`, maybe we could name it `Config` or `Settings`?

youenchene replied:
> Also instead of naming it `Options`, maybe we could name it `Config` or `Settings`?

Hello?

I've chosen options because this is the name on redocly side : https://redocly.com/docs/redoc/deployment/html#the-redoc-object

youenchene replied:
@mvrilo the`map[string]any`is legit and make it stronger.

I have to upgrade to go 1.18 to handle the `any` type.

youenchene replied:
@mvrilo if you have 5mn for the PR. It would be delightful.

youenchene replied:
@mvrilo please.

youenchene replied:
@mvrilo again pleasE.

youenchene replied:
bottle in the sea

youenchene committed:
refactor: Refactor Redoc options to use structured map instead of string.

Updated `Options` field from a JSON string to a `map[string]any` for better type safety and readability. Adjusted related logic, tests, and documentation to reflect this change. Removed redundant test for invalid JSON in options.

Jhoyola committed:
Update redoc to 2.5.0

youenchene committed:
fix: replace ioutil.ReadAll with io.ReadAll

youenchene committed:
chore: update Redoc to v2.5.1

youenchene committed:
📝 docs: add AGENTS.md with multi-module structure and dev commands
